File: app/src/main/java/in/aerem/comconbeacons/UserListItem.java
package in.aerem.comconbeacons;

import org.ocpsoft.prettytime.PrettyTime;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

public class UserListItem {
    static SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");

    public String username;
    public String location;
    public String time;

    static private String valueOr(String value, String defaultValue) {
        if (value == null || value.isEmpty())
            return defaultValue;
        return value;
    }

    public UserListItem(UsersResponse r) {
        username = r.email;
        if (r.beacon == null) {
            location = "None";
        } else {
            location = valueOr(r.beacon.label, r.beacon.bssid);
        }
        time = humanReadableDateInfo(r.updated_at);
    }

    private static String humanReadableDateInfo(String rawDate) {
        format.setTimeZone(TimeZone.getTimeZone("UTC"));
        try {
            PrettyTime p = new PrettyTime(new Locale("ru"));
            Date date = format.parse(rawDate);
            return p.format(date);
        } catch (ParseException e) {
            return "";
        }
    }
}

File: app/src/main/java/in/aerem/comconbeacons/UserListItem.kt
package `in`.aerem.comconbeacons

import `in`.aerem.comconbeacons.models.UserResponse
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverter
import java.text.SimpleDateFormat
import java.util.*

enum class Status {
    ADVENTURE,
    FREE,
    BUSY,
    UNKNOWN
}

@Entity
data class UserInfo(
    @PrimaryKey
    val id: Int,
    val username: String,
    val location: String,
    val date: Date,
    val status: Status
)

@Entity
data class UserIsFavorite(
    @PrimaryKey
    val id: Int,
    val favorite: Boolean = false
)

data class UserListItem(
    val id: Int,
    val username: String,
    val location: String,
    val date: Date,
    val status: Status,
    val favorite: Boolean = false
)

class Converters {
    @TypeConverter
    fun fromTimestamp(value: Long): Date {
        return Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date): Long {
        return date.time
    }

    @TypeConverter
    fun fromString(value: String): Status {
        return when (value) {
            "adventure" -> Status.ADVENTURE
            "free" -> Status.FREE
            "busy" -> Status.BUSY
            else -> Status.UNKNOWN
        }
    }

    @TypeConverter
    fun toString(s: Status): String {
        return s.toString().toLowerCase()
    }
}

fun fromResponse(r: UserResponse): UserInfo {
    val l = r.location;
    val location = l?.label ?: "None"

    val format = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
    format.timeZone = TimeZone.getTimeZone("Europe/Moscow")
    val date = format.parse(r.updated_at)

    return UserInfo(r.id, r.name ?: "Anonymous", location, date, Converters().fromString(r.status))
}